core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -1130,29 +1090,22 @@ void goToNext() {
 
         @Override
         public EdgeIteratorState detach(boolean reverseArg) {
-            if (edgeId == nextEdgeId || !EdgeIterator.Edge.isValid(edgeId))
-                throw new IllegalStateException("call next before detaching or setEdgeId (edgeId:" + edgeId + " vs. next " + nextEdgeId + ")");
-
-            EdgeIteratorState iter = edgeAccess.getEdgeProps(edgeId, reverseArg ? baseNode : adjNode, filter);
-            assert iter != null;
-            if (reverseArg) {
-                // for #162
-                ((EdgeIterable) iter).reverse = !reverse;
-            }
-            return iter;
+            if (edgeId == nextEdgeId)
+                throw new IllegalStateException("call next before detaching (edgeId:" + edgeId + " vs. next " + nextEdgeId + ")");
```

Comment:
BTW: I'm not sure if it is a good thing to do these logic checks via `if` instead of `assert' as they could have a potential slow down, but I wasn't able to measure this difference and so I preferred long term safety (i.e. show these problems even in production)

Reply:
Hm yes good question. An integer comparison `==` should hardly ever matter? Except the method gets too long and is no longer inlined or something? Very hard or impossible to say without measuring.